Source/ARTTypes.h

```diff
@@ -148,6 +160,9 @@ NSString *generateNonce(void);
 
 #pragma mark - ARTConnectionStateChange
 
+/**
+ A ARTConnectionStateChange is a type encapsulating state change information emitted by the Connection object. See Connection#on to register a listener for one or more events.
```

Comment:
It is, but because it was implemented as preprocessor definition, you can't see it the header. Look for ART_EMBED_INTERFACE_EVENT_EMITTER

Reply:
It would be helpful if you can find the file and link to it, or paste the code in the reply, instead of saying “look for `ART_EMBED_INTERFACE_EVENT_EMITTER`”.

Reply:
// ARTEventEmitter.h

...

// This macro adds methods to a class implementation that just bridge calls to an internal
// instance variable, which must be called _eventEmitter, of type ARTEventEmitter *.
// It's supposed to be used together with ART_EMBED_INTERFACE_EVENT_EMITTER in the
// header file of the class.
#define ART_EMBED_IMPLEMENTATION_EVENT_EMITTER(EventType, ItemType) - (ARTEventListener *)on:(EventType)event callback:(void (^)(ItemType))cb {\
return [_eventEmitter on:event callback:cb];\
}\
\
- (ARTEventListener *)on:(void (^)(ItemType))cb {\
return [_eventEmitter on:cb];\
}\
\
- (ARTEventListener *)once:(EventType)event callback:(void (^)(ItemType))cb {\
return [_eventEmitter once:event callback:cb];\
}\

Reply:
So shouldn’t the code read `ARTConnection#on`?

Reply:
It's a last fix in https://github.com/ably/ably-cocoa/commit/ff80ee88ab513d50518e33f823b70a41213d1b7f